portfolio/src/main/webapp/script.js

```diff
@@ -132,17 +132,21 @@ function randomSlide(randomButton) {
  * Fetches comments from the server and adds it to the DOM.
  */
 function getComments() {
-  const responsePromise = fetch('/data');
-  responsePromise.then(handleResponse);
+  fetch('/data').then((response) => {
+    if (!response.ok) {
+      throw new Error("Response error while fetching data: " + 
+        response.status + " (" + response.statusText + ")");
+    }
```

Comment:
This is a good way to detect the error but it will result in a unhandled promise rejection because you throw inside the asynchronous function and don't have a catch handler. Some simple way to handle this is to
 * add a `.catch(console.error)` handler that will just handle the error and print it to the console (probably good enough for now)

Also I noticed that `fetch` can itself still reject the returned promise if a network error occurs (instead of a server response that does not have HTTP status 200). See https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API (under _Differences from jQuery_). This means you'd anyway want a catch handler for that case.

Something that would work then:
```
fetch(...)
  .then((response) => {
    if (!response.ok) { throw ... }
    return response;
  })
  .then(handleResponse)
  .then(addCommentsToDom) // optinally you can even restructure this a bit so the whole control flow is visible and understandable from this function. 
 .catch((err) => {
    // handle error
  });
```

See:
https://javascript.info/promise-error-handling
https://ponyfoo.com/articles/es6-promises-in-depth#then-again

Reply:
I moved the response check inside the handleResponse, so it looks cleaner.